### PR TITLE
feat: 🎸 Added `debug` option, made rules parser debuggable

### PR DESCRIPTION
**Pull request by GerkinDev**
`debug` is now an option instead of a property, allowing to use it during construction. Grammar parsers can use a base parser with the `debug` option by using the `debugRulesParser` option. This allows the lib to be more easily debuggable.

### CODE REVIEW

File: src/Parser.ts [hunk 1]
```diff
@@ -124,14 +124,19 @@ function stripRules(token: IToken, re: RegExp) {
...
+export interface IParserOptions {
+  keepUpperRules: boolean;
+  debug: boolean;
+}
 
 const ignoreMissingRules = ['EOF'];
 
 export class Parser {
-  debug = false;
+  private readonly debug;
```
Comment (menduz) [inline src/Parser.ts]:
```suggestion
  private readonly debug: boolean;
```

Comment (menduz) [reply src/Parser.ts]: Why is it readonly now?

**Comment by GerkinDev** [comment]
Because the `debug` option is provided during construction. Do you want me to make it writable?

File: src/Parser.ts (hunk 1 above)
Comment (GerkinDev) [reply src/Parser.ts]: Because the debug option is provided during construction. But it is absolutely not required.

**Comment by GerkinDev** [comment]
Hi,

Do you have an interest in merging this PR ? Do you need anything ?

**Comment by menduz** [comment]
Sorry, I forgot to do it. You may be interested in the WebAssembly sister of this repository https://github.com/lys-lang/lys-tokenizer/blob/master/src/parser/lysgrammar.lys

**Pull request merged**